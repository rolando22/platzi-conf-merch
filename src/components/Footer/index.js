import React from "react";
import "./Footer.css";

function Footer () {
    return (
        <footer className="Footer">
            <p className="Footer-title">Platzi Conf Merch</p>
            <p className="Footer-copy">Todos los Izquierdos Reservados.</p>
        </footer>
    );
};

export { Footer };