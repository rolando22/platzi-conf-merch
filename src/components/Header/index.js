import React from "react";
import { Link } from "react-router-dom";
import "./Header.css";

function Header () {
    return (
        <header className="Header">
            <h1 className="Header-title">
                <Link to="/">
                    PlatziConf Merch
                </Link>
            </h1>
            <div className="Header-checkout">
                <Link to="/checkout">
                    Chekout
                </Link>
            </div>
        </header>
    );
};

export { Header };