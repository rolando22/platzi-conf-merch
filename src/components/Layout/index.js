import React from "react";
import { Header, Footer } from "../";
import "./Layout.css";

function Layout ({ children }) {
    return(
        <div className="Main">
            <Header />
            {children}
            <Footer />
        </div>
    );
};

export { Layout };