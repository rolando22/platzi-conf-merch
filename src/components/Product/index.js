import React from "react";
import "./Product.css";

function Product ({ product, handleAddtoCart }) {
    return (
        <div className="Product-item">
            <img
                src={product.image}
                atl={product.title}
            />
            <div className="Product-item-info">
                <h2>
                    {product.title}
                    <span>{` $${product.price}`}</span>
                </h2>
                <p>{product.description}</p>
            </div>
            <button
                type="button"
                onClick={handleAddtoCart(product)}
            >
                Comprar
            </button>
        </div>
    );
};

export { Product };