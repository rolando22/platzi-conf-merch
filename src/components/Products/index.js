import React from "react";
import { Product } from "../";
import { useAppContext } from "../../hooks/useAppContext";
import "./Products.css";

function Products () {
    const { state: { products }, addToCart } = useAppContext();

    const handleAddtoCart = (product) => () => {
        addToCart(product);
    };

    return (
        <div className="Products">
            <div className="Products-items">
                {products.map(product => (
                    <Product
                        key={product.id}
                        product={product}
                        handleAddtoCart={handleAddtoCart}
                    />
                ))}
            </div>
        </div>
    );
};

export { Products };