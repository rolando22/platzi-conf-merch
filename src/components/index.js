export { Layout } from "./Layout";
export { Header } from "./Header";
export { Footer } from "./Footer";
export { Products } from "./Products";
export { Product } from "./Product";