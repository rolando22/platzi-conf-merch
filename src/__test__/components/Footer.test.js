import React from "react";
import { shallow } from "enzyme";
import { create } from "react-test-renderer";
import { Footer } from "../../components";

describe('<Footer />', () => {
    const footer = shallow(<Footer />);

    test('Render del componente Footer', () => {
        expect(footer.length).toEqual(1);
    });
    test('Render del tÃ­tulo', () => {
        expect(footer.find('.Footer-title').text()).toEqual('Platzi Conf Merch');
    });
});

describe('Footer Snapshot', () => {
    test('Comprobar la UI del componente Footer', () => {
        const footer = create(<Footer />);
        expect(footer.toJSON()).toMatchSnapshot();
    });
});