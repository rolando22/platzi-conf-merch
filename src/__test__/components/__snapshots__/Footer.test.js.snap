// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshot Comprobar la UI del componente Footer 1`] = `
<footer
  className="Footer"
>
  <p
    className="Footer-title"
  >
    Platzi Conf Merch
  </p>
  <p
    className="Footer-copy"
  >
    Todos los Izquierdos Reservados.
  </p>
</footer>
`;
